src/npcmove.cpp

```diff
@@ -756,7 +756,7 @@ npc_action npc::address_needs(int danger)
             return npc_undecided;
         }
 
-        if( has_effect( "allow_sleep" ) || fatigue > MASSIVE_FATIGUE ) {
+        if( misc_rules.allow_sleep || fatigue > MASSIVE_FATIGUE ) {
```

Comment:
Yeah, left for compatibility. Wasn't sure how to properly remove it without hardcoding the id.